net/socks5/socks5.go

```diff
@@ -143,7 +149,8 @@ type Conn struct {
 	clientConn net.Conn
 	request    *request
 
-	udpClientAddr net.Addr
+	udpClientAddr  net.Addr
+	udpTargetConns map[socksAddr]net.Conn
```

Comment:
took me a bit to convince myself that the map access is safe and it does look like it but seems fragile. should clean it up in this PR or as a follow up.